File: MaxDepth_Nest/maxdepth_nest.js
/**
 * @param {string} s - The string cotains valid parentheses VFS
 * @returns {number} returns the max depth of nest
 */
var maxDepth = function (s) {
	let currentDepth = 0;
	let maxDepth = 0;
	for (let i = 0; i < s.length; i++) {
		if (s[i] === "(") {
			currentDepth++;

			if (currentDepth > maxDepth) {
				maxDepth = currentDepth;
			}
		}

		if (s[i] === ")") {
			currentDepth--;
		}
	}

	return maxDepth;
};

File: MaxDepth_Nest/maxdepth_nest.go
package maxdepthnest

func maxDepth(s string) int {
	maxDepth, currentDepth := 0, 0

	for _, char := range s {
		if char == '(' {
			currentDepth++
			if currentDepth > maxDepth {
				maxDepth = currentDepth
			}
		}

		if char == ')' {
			currentDepth--
		}
	}

	return maxDepth
}
